snap/snapcraft.yaml

```diff
@@ -6,6 +6,9 @@ summary: collects and exports juju machine status metrics
 description: |
   Collects juju machine status metrics periodically.
   Provides a prometheus exporter interface exposing the collected metrics.
+architectures:
+  - build-on: amd64
+  - build-on: arm64
```

Comment:
The syntax you are using is for core20 and earlier bases, which may not work for the core22 base of this snap. According to the [documentation](https://snapcraft.io/docs/architectures#heading--core22), we can simply specify it in the following format:
```
architectures:
  - amd64
  - arm64
 ```